pkg/asset/installconfig/azure/client.go

```diff
@@ -211,3 +212,60 @@ func (c *Client) ListResourceIDsByGroup(ctx context.Context, groupName string) (
 	}
 	return res, nil
 }
+
+// GetVirtualMachineSku retrieves the restource SKU of a specified virtual machine in the given region.
+func (c *Client) GetVirtualMachineSku(ctx context.Context, name, region string) (azsku.ResourceSku, error) {
+	results, err := c.GetResourceSkus(ctx, "virtualMachines", region)
+	if err != nil {
+		return azsku.ResourceSku{}, err
+	}
+
+	for _, sku := range results {
+		if strings.EqualFold(name, *sku.Name) {
+			return sku, nil
+		}
+	}
+
+	return azsku.ResourceSku{}, errors.Errorf("SKU not found for specified subscription in region %s", region)
+
+}
+
+// GetResourceSkus retrieves the resource SKUs of a specified type for a given region.
+func (c *Client) GetResourceSkus(ctx context.Context, resourceType, region string) ([]azsku.ResourceSku, error) {
+	client := azsku.NewResourceSkusClientWithBaseURI(c.ssn.Environment.ResourceManagerEndpoint, c.ssn.Credentials.SubscriptionID)
+	client.Authorizer = c.ssn.Authorizer
+	ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
+	defer cancel()
+
+	var results []azsku.ResourceSku
+
+	for page, err := client.List(ctx); page.NotDone(); err = page.NextWithContext(ctx) {
+		if err != nil {
+			return nil, errors.Wrap(err, "error fetching SKU pages")
+		}
+		for _, sku := range page.Values() {
+			// Filter out resources that are not the right type, if specified.
+			if resourceType != "" && !strings.EqualFold(resourceType, *sku.ResourceType) {
+				continue
+			}
+			// Add all resources when region is not specified.
+			if region == "" {
+				results = append(results, sku)
+				continue
+			}
+			// Only add resources in specified region.
+			for _, location := range to.StringSlice(sku.Locations) {
+				if strings.EqualFold(location, region) {
+					results = append(results, sku)
+				}
+			}
+		}
+	}
+
+	if len(results) == 0 {
+		return nil, errors.Errorf("no SKUs for specified subscription in region %s", region)
+	}
```

Comment:
Why return an error rather than just returning an empty slice and letting the caller decide how to handle there being no SKUs?

Reply:
Fixed in https://github.com/openshift/installer/commit/91402a22c47edd67c4b9802c4f2e47bfce3b3f39